### DISCUSSION TITLE
Extensions in docker-code-server template

### DISCUSSION THREAD

**Discussion by mohammedyusha** [coder/coder#5140] [General] [answered]
Hi, Can we add Vscode extensions while creating a workspace in docker-code-server template?

![image](https://user-images.githubusercontent.com/113102891/202966367-bc404605-c256-416a-99c2-2e17e0838936.png)

**Comment by jsjoeio** [comment]
Good question! Can you elaborate on what you'd want to see? Would it let you add one VS Code Extension or would you want it to be able to search and select from a list? Just trying to understand the full picture.

**Reply by ghuntley** [reply]
> Would it let you add one VS Code Extension or would you want it to be able to search and select from a list?

Hmm. I'm thinking this can be resolved through making it easier to create/modify templates without having to use the `$ coder templates` CLI. Then providing documentation/examples which demonstrate adding new extension is just a tfvar of array of stings.

mockup example below...
```tf
variable "code-server-extensions" {
  installTheseExtensions = [
    "extension1"
    "extension2"
    "extension3"
    "extension4"
   ]
}
```

Alternatively some form of integrating or documentation of how to use https://github.com/coder/code-marketplace makes sense.

**Reply by matifali** [reply]
Coder as per release 0.25.0 now doesn't support variables for workspace-scoped parameters and you need to use 
[`coder_parameter`](https://registry.terraform.io/providers/coder/coder/latest/docs/data-sources/parameter)

**Comment by matifali** [comment, marked as answer]
You can use the code-server module that supports installing extensions from a provided list or defined in the .vscode directory. 
https://registry.coder.com/modules/code-server
